netsim/cli/api.py

```diff
@@ -0,0 +1,384 @@
+      args = []
+      if payload.get("instance"):
+        args += ["--instance", str(payload.get("instance"))]
+      output_dir = payload.get("collectOutput")
+      if output_dir:
+        args += ["--output", output_dir]
+      tarball = payload.get("collectTar")
+      if tarball:
+        args += ["--tar", tarball]
+      if payload.get("collectCleanup"):
+        args.append("--cleanup")
+      _run_with_output(netlab_collect.run, args)
+    elif action == "status":
+      args = ["--all"]
+      if payload.get("instance"):
+        args = ["--instance", str(payload.get("instance"))]
+      _run_with_output(netlab_status.run, args)
+    else:
+      raise ValueError(f"unknown action {action}")
+  finally:
+    os.chdir(prev_cwd)
+
+
+def start_job(payload: Dict[str, Any]) -> Dict[str, Any]:
+  job_id = f"job-{int(time.time() * 1000)}-{os.urandom(3).hex()}"
+  log_path = os.path.join(LOG_DIR, f"{job_id}.log")
+  job = {
+    "id": job_id,
+    "action": payload.get("action") or "up",
+    "state": "queued",
+    "createdAt": now_iso(),
+    "startedAt": None,
+    "finishedAt": None,
+    "error": None,
+    "workdir": None,
+    "logPath": log_path,
+  }
+
+  def _runner() -> None:
+    job["state"] = "running"
+    job["startedAt"] = now_iso()
+    try:
+      job["workdir"] = workspace_dir(payload)
+      with open(log_path, "w", encoding="utf-8") as log_fp:
+        run_netlab_action(payload, log_fp)
+      job["state"] = "success"
+    except Exception as exc:
+      job["state"] = "failed"
+      job["error"] = f"{exc}"
+      with open(log_path, "a", encoding="utf-8") as log_fp:
+        log_fp.write("\n")
+        log_fp.write(traceback.format_exc())
+    finally:
+      job["finishedAt"] = now_iso()
+
+  thread = threading.Thread(target=_runner, daemon=True)
+  job["thread"] = thread
+  with JOB_LOCK:
+    JOBS[job_id] = job
+  thread.start()
+  return job
+
+
+def parse_json_body(handler: BaseHTTPRequestHandler) -> Dict[str, Any]:
+  length = int(handler.headers.get("Content-Length", "0") or "0")
+  if not length:
+    return {}
+  raw = handler.rfile.read(length)
+  if not raw:
+    return {}
+  return json.loads(raw.decode("utf-8"))
+
+
+def send_json(handler: BaseHTTPRequestHandler, status: int, payload: Any) -> None:
+  data = json.dumps(payload).encode("utf-8")
+  handler.send_response(status)
+  handler.send_header("Content-Type", "application/json")
+  handler.send_header("Content-Length", str(len(data)))
+  handler.end_headers()
+  handler.wfile.write(data)
+
+
+class NetlabHandler(BaseHTTPRequestHandler):
+  def log_message(self, format: str, *args: Any) -> None:
+    return
+
+  def do_GET(self) -> None:
+    if not require_auth(self):
+      return
+    parsed = urllib.parse.urlparse(self.path)
+    parts = [p for p in parsed.path.strip("/").split("/") if p]
+    if not parts:
+      send_json(self, HTTPStatus.NOT_FOUND, {"error": "not found"})
+      return
+
+    if parts == ["healthz"]:
```

Comment:
I would usually implement something like this with a dictionary mapping keywords (`parts` in you case) into functions that do the actual work.